ibis/backends/flink/__init__.py

```diff
@@ -635,6 +638,134 @@ def drop_view(
         sql = statement.compile()
         self._exec_sql(sql)
 
+    def _read_file(
+        self,
+        file_type: str,
+        path: str | Path,
+        schema: sch.Schema,
+        table_name: str | None = None,
+    ) -> ir.Table:
+        """Register a file as a table in the current database.
+
+        Parameters
+        ----------
+        file_type
+            File type, e.g., parquet, csv, json.
+        path
+            The data source.
+        schema
+            The schema for the new table.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+
+        Raises
+        ------
+        IbisError
+            If `schema` is None.
+        """
+        if schema is None:
+            raise exc.IbisError("`schema` must be set while calling `_read_file()`")
```

Comment:
```suggestion
            raise ValueError(f"`schema` must be explicitly provided when calling `read_{file_type}`")
```

Reply:
Thanks for the improvements. Incorporated the suggested changes.